chalice/app.py

```diff
@@ -450,7 +516,11 @@ def __call__(self, event, context):
                                                     function_args)
         if self._cors_enabled_for_route(route_entry):
             self._add_cors_headers(response, route_entry.cors)
-        return response.to_dict()
+        if self._matches_content_type(response.case_insensitive_headers.get(
```

Comment:
I don't see the _matches_content_type being tested with response headers.  If it isn't tested, can you add something that verifies headers with the `; extra-stuff` appended will still match appropriately?